app/gui/view/component-browser/component-list-panel/src/button_panel.rs

```diff
@@ -0,0 +1,211 @@
+        let shortcuts_tooltip = tooltip(SHOW_SHORTCUTS_TOOLTIP);
+        let shortcuts = new_button::<icon::command_key::Shape>(app, shortcuts_tooltip);
+        let unstable_tooltip = tooltip(UNSTABLE_TOOLTIP);
+        let unstable = new_button::<icon::unstable::Shape>(app, unstable_tooltip);
+        let marketplace_tooltip = tooltip(MARKETPLACE_TOOLTIP);
+        let marketplace = new_button::<icon::marketplace::Shape>(app, marketplace_tooltip);
+        let side_panel_tooltip = tooltip(SIDE_PANEL_TOOLTIP);
+        let side_panel = new_button::<icon::right_side_panel::Shape>(app, side_panel_tooltip);
+        let size = Vector2(icon::SIZE, icon::SIZE);
+        local_scope.set_size(size);
+        shortcuts.set_size(size);
+        unstable.set_size(size);
+        marketplace.set_size(size);
+        side_panel.set_size(size);
+        background.add_child(&local_scope);
+        background.add_child(&shortcuts);
+        background.add_child(&unstable);
+        background.add_child(&marketplace);
+        background.add_child(&side_panel);
+
+        Self { background, local_scope, shortcuts, unstable, marketplace, side_panel }
+    }
+
+    fn update_style(&self, style: &Style) {
+        self.background.set_size(Vector2(style.width, style.height));
+        self.background.set_corner_radius(style.corner_radius);
+        self.background.set_inner_border(style.border_width, style.inner_border_distance);
+        self.background.set_border_color(style.border_color);
+        self.background.set_color(style.background_color);
+        self.background.set_gap((style.gap, 0.0)).set_padding_all(style.padding);
+        self.side_panel.set_margin_left(style.right_side_margin);
+    }
+}
+
+
+
+// ============
+// === View ===
+// ============
+
+/// A narrow bar on the top of the Component List Panel that contains various buttons for
+/// controlling the Component Browser.
+#[derive(Debug, Clone, CloneRef, Deref)]
+pub struct View {
+    model: Model,
+    #[deref]
+    frp:   Frp,
+}
+
+impl View {
+    pub fn new(app: &Application) -> Self {
+        let model = Model::new(app);
+        let frp = Frp::new();
+
+        let network = frp.network();
+        let out = &frp.private.output;
+
+        let styles = StyleWatchFrp::new(&scene().style_sheet);
+        let style = Style::from_theme(network, &styles);
+        frp::extend! { network
+            init <- source_();
+            eval style.update((style) model.update_style(style));
+
+
+            // === Initial state ===
+
+            model.side_panel.set_state <+ init.constant(true);
+            // Buttons below are not implemented.
+            model.local_scope.set_read_only <+ init.constant(true);
+            model.shortcuts.set_read_only <+ init.constant(true);
+            model.unstable.set_read_only <+ init.constant(true);
+            model.marketplace.set_read_only <+ init.constant(true);
```

Comment:
Instead of using `init.constant`, the values can be emitted directly. This saves on creating multiple unnecessary FRP nodes and creating connections with them.
```suggestion
            model.side_panel.set_state(true);
            // Buttons below are not implemented.
            model.local_scope.set_read_only(true);
            model.shortcuts.set_read_only(true);
            model.unstable.set_read_only(true);
            model.marketplace.set_read_only(true);
```